Src/Newtonsoft.Json/Linq/JValue.Async.cs

```diff
@@ -0,0 +1,145 @@
+                    if (_value is int)
+                    {
+                        return writer.WriteValueAsync((int)_value, cancellationToken);
+                    }
+
+                    if (_value is long)
+                    {
+                        return writer.WriteValueAsync((long)_value, cancellationToken);
+                    }
+
+                    if (_value is ulong)
+                    {
+                        return writer.WriteValueAsync((ulong)_value, cancellationToken);
+                    }
+
+#if !(PORTABLE) || NETSTANDARD1_1
+                    if (_value is BigInteger)
+                    {
+                        return writer.WriteValueAsync((BigInteger)_value, cancellationToken);
+                    }
+#endif
+
+                    return writer.WriteValueAsync(Convert.ToInt64(_value, CultureInfo.InvariantCulture), cancellationToken);
+                case JTokenType.Float:
+                    if (_value is decimal)
+                    {
+                        return writer.WriteValueAsync((decimal)_value, cancellationToken);
+                    }
+
+                    if (_value is double)
+                    {
+                        return writer.WriteValueAsync((double)_value, cancellationToken);
+                    }
+
+                    if (_value is float)
+                    {
+                        return writer.WriteValueAsync((float)_value, cancellationToken);
+                    }
+
+                    return writer.WriteValueAsync(Convert.ToDouble(_value, CultureInfo.InvariantCulture), cancellationToken);
+                case JTokenType.String:
+                    return writer.WriteValueAsync(_value != null ? _value.ToString() : null, cancellationToken);
+                case JTokenType.Boolean:
+                    return writer.WriteValueAsync(Convert.ToBoolean(_value, CultureInfo.InvariantCulture), cancellationToken);
+                case JTokenType.Date:
+                    if (_value is DateTimeOffset)
+                    {
+                        return writer.WriteValueAsync((DateTimeOffset)_value, cancellationToken);
+                    }
+
+                    return writer.WriteValueAsync(Convert.ToDateTime(_value, CultureInfo.InvariantCulture), cancellationToken);
+                case JTokenType.Bytes:
+                    return writer.WriteValueAsync((byte[])_value, cancellationToken);
+                case JTokenType.Guid:
+                    return writer.WriteValueAsync(_value != null ? (Guid?)_value : null, cancellationToken);
+                case JTokenType.TimeSpan:
+                    return writer.WriteValueAsync(_value != null ? (TimeSpan?)_value : null, cancellationToken);
+                case JTokenType.Uri:
+                    return writer.WriteValueAsync((Uri)_value, cancellationToken);
+            }
+
+            throw MiscellaneousUtils.CreateArgumentOutOfRangeException("TokenType", _valueType, "Unexpected token type.");
```

Comment:
There's no such argument, and as such this probably shouldn't have been an `ArgumentOutOfRangeException` but `InvalidOperationException` or something similar. I'm leaving it as-is.